assembl/views/admin/views.py

```diff
@@ -18,12 +19,79 @@
 from assembl.auth.util import (
     add_multiple_users_csv, user_has_permission, get_permissions)
 from assembl.models.auth import (
-    create_default_permissions, User, Username, AgentProfile)
+    create_default_permissions, User, Username, AgentProfile,
+    LanguagePreferenceOrder)
+from assembl.models import Preferences, Locale
+from assembl import locale_negotiator
+from assembl.lib.utils import get_global_base_url
+from assembl.nlp.translation_service import DummyGoogleTranslationService
+from ..backbone.views import process_locale
 
 
 _ = TranslationStringFactory('assembl')
 
 
+class PseudoDiscussion(object):
+    id = 0
+    topic = "Administration"
+    slug = "admin"
+    homepage_url = None
+    logo = None
+    def translation_service(self):
+        return None
+    def get_base_url(self, *args):
+        return get_global_base_url(True)
+    def get_url(self, *args):
+        return get_global_base_url(True)
+    def get_all_agents_preload(self, user):
+        return []
+
+
+@view_config(route_name='base_admin', request_method='GET', http_cache=60,
+             permission=P_SYSADMIN)
+def base_admin_view(request):
+    """The Base admin view, for frontend urls"""
+    user_id = authenticated_userid(request) or Everyone
+    context = get_default_context(request)
+
+    session = Discussion.default_db
+    preferences = Preferences.get_default_preferences(session)
+    user = User.get(user_id)
+
+    if '_LOCALE_' in request.cookies:
```

Comment:
Very unDRY. I'll make this a method, as it's being used a few view_config's now..